### PR TITLE
Entities/lecture

### PR DESCRIPTION
**Pull request by JohaJung**
This PR introduces the backend parts of a lecture entity, so it resolves the backend part of #34 

I found a very nice way to reduce boilerpate code for all those entities, all REST endpoints are auto-generated based on the URL schema explained here (replace `invoices` by `lectures`:
https://javatechonline.com/how-to-reduce-boilerplate-code-in-rest-api-spring-data-rest#Step_3_Create_URL_for_CRUD_operations

We should discuss tomorrow whether to add backend tests later or in this PR...

---

### Implementation

- [x] Does this code change accomplish what it is supposed to do?
- [ ] Can this solution be simplified without unreasonable effort
- [x] Is the code modular enough?
- [ ] Does similar functionality already exist in the codebase? If yes, why isn’t it reused?
- [ ] Single Responsibility Principle: Does any class serve only one purpose? If not, it should probably split up in multiple classes.
- [x] Interface Segregation: If interface implemented, is it fully implemented?
- [x] Dependency Injection: Are central object instantiation processes managed by a single context?
- [x] If any functionality is missing, is it marked with a comment that starts with TODO:?
- [ ] Do you see any potential to improve the performance of the code significantly? 

### Logic Errors and Bugs

- [ ] Can you think of any use case in which the code does not behave as intended?
- [ ] Can you think of any inputs or external events that could break the code?
- [ ] Are edge cases covered by tests?

### Error Handling and Logging

- [x] Is error handling done the correct way?
- [ ] Should any logging or debugging information be added or removed?
- [x] Are error messages user-friendly?

### Dependencies (internal and external)

- [ ] Were updates to documentation, configuration, or readme files made as required by this change?
- [ ] Is the API.md up-to-date?
- [ ] Are the enumerations between frontend and backend synchronized?

### Security and Data Privacy

- [ ] Does the code introduce any security vulnerabilities?
- [ ] Is (user) input validated, sanitized, and escaped to prevent security attacks such as cross-site scripting or SQL injection?
- [ ] Is sensitive data like user data handled and stored securely?
- [ ] Is the right encryption used?
- [ ] Is data retrieved from external APIs or libraries checked for security issues?
- [ ] Were any secrets committed?

### Testing and Testability (later)

- [ ] Have automated tests been added, or have related ones been updated to cover the change?
- [ ] Do the existing tests reasonably cover the code change?
- [ ] Are there some test cases, input or edge cases that should be tested in addition?

### Usability and Accessibility

- [ ] Is the proposed solution (UI) accessible (e.g. colorblindness, ...)?
- [ ] Is the API/UI intuitive to use?
- [ ] Was the app tested on another device?

### Readability

- [x] Is the code easy to understand?
- [ ] Which parts were confusing to you and why?
- [x] Can the readability of the code be improved by smaller methods/different names?
- [ ] Is the code located in the right file/folder?
- [ ] Do you think certain methods should be restructured to have a more intuitive control flow?
- [ ] Are there redundant or outdated comments?
- [ ] Could some comments convey the message better?
- [ ] Could some comments be removed by making the code itself more readable?

### CODE REVIEW

**Comment by JohaJung** [comment]
This pull request is backend-wise in a state I'm willing to go forward with for the other entities. So I'd appreciate reviews. ( @polodasi123 the other two PRs you reviewed yesterday will be adapted in the same manner. Thus I recommend reviewing this one to see the desing I chose for the backend entities.)

This also includes a working proof of concept for the frontend creating entities in the backend via the API. Everything further for the frontend should be done in a separate PR.

File: VLA-Frontend/src/hooks/useLectures.ts [hunk 1]
```diff
@@ -1,20 +1,41 @@
 import { useState } from "react";
 import type { Lecture } from "../components/calendar/CalendarTypes";
 
+// TODO: move to some central file
```
[the remark this answers is not in the document]
Comment (MarcoLeipnitz) [reply VLA-Frontend/src/hooks/useLectures.ts]: Ah sorry, this is the frontend, factor this out to one central api.ts file or something comparable.
Maybe we can inject this into the frontend somehow?

File: VLA-Frontend/src/hooks/useLectures.ts [hunk 2]
```diff
@@ -1,20 +1,41 @@
...
+      method: "POST",
+      headers: headers,
+      body: JSON.stringify(lecture),
+    });
+    fetch(request).then(res => {
+      console.log("response from adding lecture:");
+      res.json().then(data => {
+        console.log(data);
+        setLectures((prev) => [...prev, data as Lecture]);
+      });
```
Comment (MarcoLeipnitz) [inline VLA-Frontend/src/hooks/useLectures.ts]: You are not catching errors that might occur during the creation process

File: VLA-Backend/src/main/java/de/vlaorgatu/vlabackend/calendar/lecture/LectureController.java [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+    private final LectureRepository lectureRepository;
+
+    @PostMapping("/lectures")
+    public ResponseEntity<?> createLecture(@RequestBody Lecture lecture) {
+        if (Objects.nonNull(lecture.getId())) {
+            return ResponseEntity.badRequest().build();
+        }
+
+        Lecture savedLecture = lectureRepository.save(lecture);
+        SseController.notifyDebugTest("Lecture created: " + savedLecture);
```
[the remark this answers is not in the document]
Comment (JohaJung) [reply VLA-Backend/src/main/java/de/vlaorgatu/vlabackend/calendar/lecture/LectureController.java]: Yes, this needs to be adjusted once the architecture how a SSE event triggers an update n the frontend is clear. (will add a TODO comment)

**Review by MarcoLeipnitz — APPROVED** [review]

**Comment by JohaJung** [comment]
Plz review, should be ready to go :)
